enhancements/machine-config/certificate-authorities-for-image-registries.md

```diff
@@ -0,0 +1,185 @@
+---
+title: certificate-authorities-for-image-registries
+authors:
+  - dmage
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2022-12-01
+last-updated: 2022-12-01
+tracking-link:
+  - https://issues.redhat.com/browse/IR-230
+see-also:
+replaces:
+superseded-by:
+---
+
+# Additional trusted certificate authorities for image registries
+
+## Summary
+
+This enhancement describes how certificate authorities for image registries
+should be distributed to CRI-O, openshift-apiserver, and other clients when the
+image registry operator is not installed.
+
+## Motivation
+
+The image registry operator manages a daemon set that distributes certificate
+authorities for the integrated image registry, and it also distributes
+certificate authorities for external registries that are provided by the
+cluster administrator in the image config. This makes the image registry
+operator required for the cluster while the integrated image registry is
+optional and can be removed.
+
+The goal of this enhancement is to move this required functionality to an
+essential cluster operator, and have the image regsitry operator only manage
+the optional registry workload.
+
+### User Stories
+
+* As a <role>, I want to <take some action> so that I can <accomplish a
+goal>.
+
+### Goals
+
+* Make additionalTrustedCA be managed by machine-config-operator.
+* Create a mechanism for the cluster-image-registry-operator to distribute
+  certificate authorities for the integrated image registry.
+* Remove the node-ca daemon set.
+
+### Non-Goals
+
+* Change additionalTrustedCA behavior for its users.
+
+## Proposal
+
+Create a controller inside cluster-config-operator that will replace the
+node-ca daemon set.
+
+This controller should handle and distribute to all nodes certificate
+authorities from the user-provided config map that is specified in
+`images.config.openshift.io/cluster`. It should also observe a config map in
+the `openshift-config-managed` namespace and merge its content with the
+user-provided config map. This will allow cluster-image-registry-operator to
+provide certificate authorities for the integated image registry without
+changing the user-provided config map.
```

Comment:
Unsaid in the current proposal, but implied by the `enhancements/machine-config/` position, is that the output of this controller will be MachineConfigs feeding into whichever MachineConfigPools are necessary to get this new content rolled out over the whole cluster.  Can we spell that out in detail here, possibly with feedback from the MachineConfig folks about how to target whichever MachineConfigPools are required?

Reply:
Yes, I think this would generate MachineConfig.